proctor-consumer/src/main/java/com/indeed/proctor/consumer/AbstractGroupsManager.java

```diff
@@ -83,26 +96,56 @@ protected ProctorResult determineBucketsInternal(final Identifiers identifiers,
                     emptyMap()
             );
         }
-        final ProctorResult proctorResult = proctor.determineTestGroups(identifiers, context, forcedGroups);
+        final ProctorResult proctorResult = proctor.determineTestGroups(
+                identifiers,
+                context,
+                forceGroupsOptions,
+                emptyList()
+        );
 
         interceptor.afterDetermineGroups(proctorResult);
         return proctorResult;
     }
 
+    /**
+     * @deprecated Use {@link #determineBucketsInternal(HttpServletRequest, HttpServletResponse, Identifiers, Map, boolean)}
+     */
+    @Deprecated
+    @VisibleForTesting
+    protected ProctorResult determineBucketsInternal(
+            final Identifiers identifiers,
+            final Map<String, Object> context,
+            final Map<String, Integer> forcedGroups
+    ) {
+        return determineBucketsInternal(
+                identifiers,
+                context,
+                ForceGroupsOptions.builder()
+                        .putAllForceGroups(forcedGroups)
+                        .build()
+        );
+    }
+
+
     protected abstract Map<String, TestBucket> getDefaultBucketValues();
 
     /**
      * servlet-based-application friendly version of determineBucketsInternal, also enabling forcing groups via request headers and cookies
      */
-    protected ProctorResult determineBucketsInternal(final HttpServletRequest request, final HttpServletResponse response, final Identifiers identifiers,
-            final Map<String, Object> context, final boolean allowForcedGroups) {
-        final Map<String, Integer> forcedGroups;
+    protected ProctorResult determineBucketsInternal(
+            final HttpServletRequest request,
+            final HttpServletResponse response,
+            final Identifiers identifiers,
+            final Map<String, Object> context,
+            final boolean allowForcedGroups
+    ) {
+        final ForceGroupsOptions forcedGroups;
```

Comment:
nits: I'd rename this `forceGroupsOptions`.

Reply:
Sounds good. Updated https://github.com/indeedeng/proctor/pull/64/commits/742674c20858b312a205aa19efdc78853bc6621d